mypyc/genops.py

```diff
@@ -2437,8 +2525,33 @@ def visit_var(self, o: Var) -> None:
     def visit_yield_from_expr(self, o: YieldFromExpr) -> Value:
         raise NotImplementedError
 
-    def visit_yield_expr(self, o: YieldExpr) -> Value:
-        raise NotImplementedError
+    def visit_yield_expr(self, expr: YieldExpr) -> Value:
+
+        self.goto_new_block()
+
+        if expr.expr:
+            retval = self.accept(expr.expr)
+            retval = self.coerce(retval, self.ret_types[-1], expr.line)
+        else:
+            retval = self.add(PrimitiveOp([], none_op, line=-1))
+
+        # Increment the next label value by 1, so that the next time '__next__' is called, we
+        # continue at the label immediately following this yield expression.
+        generator_class = self.fn_info.generator_class
+        line = self.fn_info.fitem.line
+
+        next_block = BasicBlock()
+        next_label = len(generator_class.blocks)
+        generator_class.blocks.append(next_block)
+        self.assign_to_target(generator_class.next_label_target,
+                              self.add(LoadInt(next_label)),
+                              line)
+        # TODO: Should this be INVALID_VALUE?
+        value = self.add(Return(retval))
```

Comment:
By the python semantics, the value produced should generally be `None`, I think.